### PR TITLE
#622 Hot reloading on https

### LINKED ISSUE #622: Hot reloading on https
Description:
  'http://localhost:8080/js/app.js'. This request has been blocked; the content must be served over HTTPS.

Steps To Reproduce:

1. Valet Secure
2. npm run hot
3. access the site

### PR DESCRIPTION
**Pull request by apasov**
Fixes #622. Describe in the docs how to use hot reloading on https.

### CODE REVIEW

File: docs/hot-module-replacement.md [hunk 1]
```diff
@@ -43,6 +43,33 @@ However, it can be a burden to manually change this URL for production deploys.
 With this adjustment, Laravel will do the work for you. If you run 'npm run hot' to enable hot reloading, the function will set the necessary `http://localhost:8080` base url. If, instead, you use `npm run dev` or `npm run production`, it'll use your domain as the base.
 
 
+### Usage on HTTPS
+
+If you develop your app on a HTTPS connection your hot realoading scripts and styles must also be served via HTTPS. To achieve this add `--https` option to the `hot` option command in package.json:
+
+```js
+  "scripts": {
+    "hot": "cross-env NODE_ENV=development webpack-dev-server --inline --hot --https",
```
Comment (ruchern) [inline docs/hot-module-replacement.md]: Remove `cross-env` here.

File: docs/hot-module-replacement.md [hunk 2]
```diff
@@ -43,6 +43,33 @@ However, it can be a burden to manually change this URL for production deploys.
...
+  }
+```
+
+With the above setting the `webpack-dev-server` will generate a self-signed certificate for you.
+If you want to use your own certificate use these settings:
+
+```js
+    "hot": "cross-env NODE_ENV=development webpack-dev-server --inline --hot --https --key /path/to/server.key --cert /path/to/server.crt --cacert /path/to/ca.pem",
```
Comment (ruchern) [inline docs/hot-module-replacement.md]: Remove `cross-env` here. Got to be consistent with Laravel ^1.*.

**Comment by apasov** [comment]
Really don't understand why cross-env must be removed. It's default Laravel setup. Probably you'd better propose removing it at laravel repo: https://github.com/laravel/laravel/blob/master/package.json

And why exactly here?
Above in this very doc you can see:
```js
  "scripts": {
    "dev": "cross-env NODE_ENV=development webpack --progress --hide-modules",
    "watch": "cross-env NODE_ENV=development webpack --watch --progress --hide-modules",
    "hot": "cross-env NODE_ENV=development webpack-dev-server --inline --hot",
    "production": "cross-env NODE_ENV=production webpack --progress --hide-modules"
  }
```
All the options start with `cross-env`

**Comment by ruchern** [comment]
@apasov https://github.com/JeffreyWay/laravel-mix/releases/tag/v1.0.0

>cross-env has been removed as a dependency of Mix. It doesn't make sense for us to maintain it. Frameworks like Laravel will already pull in cross-env, but for your personal projects, either open your package.json file and remove the cross-env path from all of your npm scripts, or do npm install cross-env.

**Comment by ruchern** [comment]
I also want to see this gets merged, so just letting you know that with this, I think it's likely to get merged.

**Pull request merged**